spec/models/mef_spec.rb

```diff
@@ -93,4 +93,20 @@
       end
     end
   end
+
+  describe "with_wage scope" do
+    let(:mefs) { create_list(:mef, 4) }
+    let(:wages) { Wage.where(id: mefs.map(&:wage).pluck(:id)) }
+    let(:mefs_collection) { described_class.where(id: mefs.pluck(:id)) }
+
+    before do
+      wages[0].update(mef_codes: [mefs[0].code, "123", "456"])
+      wages[1].update(mef_codes: ["123", mefs[1].code, "456"])
+      wages[2].update(mef_codes: ["123", "456", mefs[2].code])
+    end
+
+    it "joins the correct wages" do
+      expect(mefs_collection.with_wages.pluck(:"wages.id").sort).to eq wages.pluck(:id).sort
```

Comment:
au lieu de test avec un sort tu peux utiliser

`expect(collection).to contain_exactly(autre_collection)`